src/main/java/nextstep/subway/application/LineService.java

```diff
@@ -20,65 +21,62 @@ public class LineService {
 
     private LineRepository lineRepository;
 
+    private SectionRepository sectionRepository;
+
     private StationRepository stationRepository;
 
     public LineService(LineRepository lineRepository,
+                       SectionRepository sectionRepository,
                        StationRepository stationRepository) {
         this.lineRepository = lineRepository;
+        this.sectionRepository = sectionRepository;
         this.stationRepository = stationRepository;
     }
 
     @Transactional
     public LineResponse saveLine(LineRequest lineRequest) {
         Station upStation = getStation(lineRequest.getUpStationId());
         Station downStation = getStation(lineRequest.getDownStationId());
-        Line line = lineRepository.save(lineRequest.toLine(upStation, downStation));
-        return LineResponse.of(line);
+
+        Sections sections = new Sections();
+        sections.add(sectionRepository.save(
+                upStation.toSection(downStation, lineRequest.getDistance())));
+
+        return lineRepository.save(lineRequest.toLine(sections)).toLineResponse();
     }
 
     @Transactional(readOnly = true)
     public LineResponse findById(Long id) {
-        return LineResponse.of(getLine(id));
+        return getLine(id).toLineResponse();
     }
 
     @Transactional(readOnly = true)
     public List<LineResponse> findAllLines() {
         return lineRepository.findAll()
                 .stream()
-                .map(LineResponse::of)
+                .map(Line::toLineResponse)
                 .collect(Collectors.toList());
     }
 
     @Transactional
     public LineResponse updateById(Long id, LineRequest lineRequest) {
         Line line = getLine(id);
-        return LineResponse.of(line.updateInfo(lineRequest));
+        return line.updateByLineRequest(lineRequest).toLineResponse();
```

Comment:
```suggestion
        return line.update(lineRequest.toLine(null)).toLineResponse();
```
이렇게 하면 `Line`에서 `LineRequest`를 의존하는걸 피할수 있을것 같은데 어떻게 생각하시나요? 😀

Reply:
위와 같은 상황이네요!? 도메인 모델이 DTO를 의존하는 방향을 지양하자는 말씀이신거죠?! 🤔